vowpalwabbit/label_parser.h

```diff
@@ -12,19 +12,20 @@
 
 struct parser;
 struct shared_data;
+struct polylabel;
 
 struct label_parser
 {
-  void (*default_label)(void*);
-  void (*parse_label)(parser*, shared_data*, void*, v_array<VW::string_view>&);
-  void (*cache_label)(void*, io_buf& cache);
-  size_t (*read_cached_label)(shared_data*, void*, io_buf& cache);
-  void (*delete_label)(void*);
-  float (*get_weight)(void*);
-  void (*copy_label)(void*, void*);  // copy_label(dst,src) performs a DEEP copy of src into dst (dst is allocated
+  void (*default_label)(polylabel*);
+  void (*parse_label)(parser*, shared_data*, polylabel*, v_array<VW::string_view>&);
+  void (*cache_label)(polylabel*, io_buf& cache);
+  size_t (*read_cached_label)(shared_data*, polylabel*, io_buf& cache);
+  void (*delete_label)(polylabel*);
+  float (*get_weight)(polylabel*);
+  void (*copy_label)(polylabel*, polylabel*);  // copy_label(dst,src) performs a DEEP copy of src into dst (dst is allocated
```

Comment:
copying will effectively clear the label if both polylabel* end up pointing to the same thing (based on current impl of copy_array). Not sure if we should protect from this error on the api level.

Reply:
Hmm that's interesting, I think I'm going to hold off on an API fix for that because I have another branch lined up that will fix the copy behavior of v_array.